examples/raspberry-pi-threads.rs

```diff
@@ -0,0 +1,57 @@
+use navigator_rs::{Navigator, UserLed};
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread::{self, sleep};
+use std::time::Duration;
+
+fn main() {
+    println!("Creating your SensorData cache!");
+    let sensor_data = Arc::new(RwLock::new(navigator_rs::SensorData::default()));
+
+    println!("Creating your navigator module!");
+    let nav = Arc::new(Mutex::new(Navigator::new()));
+
+    println!("Setting up your navigator, ahoy!");
+    nav.lock().unwrap().init();
+
+    // This code block creates a thread that updates the `sensor_data` periodically.
+    let sensor_data_cloned = sensor_data.clone();
+    let nav_cloned = nav.clone();
+    let _ = thread::spawn(move || {
+        loop {
+            if let Ok(mut sensor_data) = sensor_data_cloned.write() {
+                *sensor_data = nav_cloned.lock().unwrap().read_all();
+                // println!("Updated value: {sensor_data:?}");
```

Comment:
why not uncomment this?

Reply:
it was a bit annoying when testing o/
uncommented;